### ISSUE TITLE
[FR] Add new argument r/tfe_workspace: drift_detection

### ISSUE THREAD

**Issue by vlsokol** [hashicorp/terraform-provider-tfe#542]
#### Use-cases
https://www.hashicorp.com/blog/terraform-cloud-adds-drift-detection-for-infrastructure-management

#### Proposal
New argument for TFE to enable/disable drift detection: (https://www.hashicorp.com/blog/terraform-cloud-adds-drift-detection-for-infrastructure-management)

```
resource "tfe_workspace" "test" {
  name            = "my-workspace-name"
  organization    = tfe_organization.test-organization.name
  drift_detection = true
}
```

**Comment by ausfestivus** [comment]
Probably also worth mentioning that the `r/tfe_notification_configuration` resource will also need to add support for the new triggers available under Notifications now:

![Screen Shot 2022-07-12 at 10 22 11 pm](https://user-images.githubusercontent.com/40222285/178488550-2cce6523-dfcf-4b0f-bd64-299c6543f19b.png)

**Comment by rhughes1** [comment]
Looks like you can do it already?

Workspace:
https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace#assessments_enabled

Notification Configuration
https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/notification_configuration#triggers

**Comment by brandonc** [comment]
@rhughes1 Thank you! Indeed this was added in [v0.37.0](https://github.com/hashicorp/terraform-provider-tfe/blob/main/CHANGELOG.md#v0380-october-24-2022) (but v0.38.0 contained a critical bug fix for assessments_enabled)

**Comment by brandonc** [comment]
I was slightly hasty in my response. The new notification triggers were released in v0.39.0
